File: types/flight-data.ts
export interface FlightData {
  timestamp: string;
  eta: string | null;
  flightDuration: number;
  flightNumber: string;
  latitude: number;
  longitude: number;
  noseId: string;
  paState: string | null;
  vehicleId: string;
  destination: string;
  origin: string;
  flightId: string;
  airspeed: number | null;
  airTemperature: number | null;
  altitude: number;
  distanceToGo: number;
  doorState: string;
  groundspeed: number;
  heading: number;
  timeToGo: number;
  wheelWeightState: string;
  grossWeight: number | null;
  windSpeed: number | null;
  windDirection: number | null;
  flightPhase: string;
} 

File: Inflight Tracker/Models/FlightData.swift
import Foundation

struct FlightData: Codable, Equatable {
    let timestamp: String
    let eta: String?
    let flightDuration: Double
    let flightNumber: String
    let latitude: Double
    let longitude: Double
    let noseId: String
    let paState: String?
    let vehicleId: String
    let destination: String
    let origin: String
    let flightId: String
    let airspeed: Double?
    let airTemperature: Double?
    let altitude: Double
    let distanceToGo: Double?
    let doorState: String?
    let groundspeed: Double
    let heading: Double?
    let timeToGo: Double
    let wheelWeightState: String
    let grossWeight: Double?
    let windSpeed: Double?
    let windDirection: Double?
    let flightPhase: String
}